File: code/text_data/first_viz/first_viz.py
#!/usr/bin/env python
# encoding: utf-8
"""
first_viz.py

Purpose:  Creating a first matplotlib visualization
            (Strata - Data Bootcamp Tutorial)
Author:   Drew Conway
Email:    drew.conway@nyu.edu
Date:     2011-01-25

Copyright (c) 2011, under the Simplified BSD License.  
For more information on FreeBSD see: http://www.opensource.org/licenses/bsd-license.php
All rights reserved.
"""

import sys
import os
import matplotlib.pylab as plt
from scipy.stats import norm

def plot_normal(random_numbers, path=""):
    """
    A function to graphically check a random distribution's
    fit to a theoretical normal.
    """
    fig=plt.figure(figsize = (8,6)) # Create a figure to plot in (good habit)
    # Histogram of random numbers with 25 bins
    n, bins, pataches = plt.hist(random_numbers,normed=True,bins=25,alpha=0.75)  
    # Add "best fit" line
    y = norm.pdf(bins)
    plt.plot(bins,y,"r-")
    # Save plot
    plt.xlabel("Random numbers")
    plt.ylabel("Density")
    plt.title("My first matplotlib visualization!")
    plt.savefig(path)

def main():
    random_normal = norm.rvs(0,1,size=10000)
    plot_normal(random_normal,"../../../images/figures/matplotlib_first.png")

if __name__ == '__main__':
	main()

File: code/text_data/first_viz/first_viz.R
# File-Name:       first_viz.R           
# Date:            2011-01-25                                
# Author:          Drew Conway
# Email:           drew.conway@nyu.edu                                      
# Purpose:         Creating a first matplotlib visualization
#                   (Strata - Data Bootcamp Tutorial)
# Data Used:       
# Packages Used:   ggplot2
# Output File:     ggplot2_first.png
# Data Output:     
# Machine:         Drew Conway's MacBook Pro

# Copyright (c) 2011, under the Simplified BSD License.  
# For more information on FreeBSD see: http://www.opensource.org/licenses/bsd-license.php
# All rights reserved.                                                         

# Load libraries
library(ggplot2)

# Generate random numbers and create data frame
random.numbers <- rnorm(10000,0,1)
norm.dframe <- as.data.frame(list(Norm=random.numbers))

# Build graphical layers
norm.plt <- ggplot(norm.dframe,aes(Norm))+ # The first layer is the ggplot2 object, and a data frame
    geom_histogram(aes(y = ..density.., fill="blue", colour="black",alpha=0.75))+    # Build a histogram
    stat_function(fun = dnorm, colour = "red")+     # Overlay the normal density function
    scale_colour_manual(values = c("black"="black","red"="red"), legend = FALSE)+   # Do housekeeping on colors
    scale_fill_manual(values = c("blue"="blue"), legend = FALSE)+
    scale_alpha(legend = FALSE)+
    xlab("Random numbers")+ylab("Density")+     # Add labels
    opts(title="My first ggplot2 visualization!")
    
# Save plot
ggsave(plot = norm.plt, filename = "../../../images/figures/ggplot2_first.pdf", height = 6, width = 8)